CRM/Contribute/Selector/Search.php

```diff
@@ -412,6 +412,11 @@ public function &getRows($action, $offset, $rowCount, $sort, $output = NULL) {
       $row['contribution_status_name'] = CRM_Utils_Array::value($row['contribution_status_id'],
         $contributionStatuses
       );
+      $invoiceSettings = Civi::settings()->get('contribution_invoice_settings');
```

Comment:
This will need to change but hopefully we can separate the Search from the UserDashboard since we don't want to be doing this on all search results

Reply:
There is a big obstruction in implementing this separation as both the Search and UserDashboard uses the same Selector to build search query and assign its result to template. Unless if we can decide the returnproperties / select elements for respective forms, something like
```patch
$controller = new CRM_Core_Controller_Simple(
      'CRM_Contribute_Form_Search',
      ts('Contributions'),
      NULL,
      FALSE, FALSE, TRUE, FALSE
    );
    $controller->setEmbedded(TRUE);
    $controller->reset();
    $controller->set('limit', 12);
    $controller->set('cid', $this->_contactId);
 +   $controller->set('returnproperties', [
 +       'contact_id',
 +   'contribution_id',
 +   'contact_type',
 +   'sort_name',
 +   'amount_level',
 +   'total_amount',
 +   'financial_type',
  +   'balance',
  +    ]);
    $controller->set('context', 'user');
    $controller->set('force', 1);
    $controller->process();
    $controller->run();
```
At [here](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/Page/UserDashboard.php#L49)

Reply:
@monishdeb there is another PR here https://github.com/civicrm/civicrm-core/pull/13147 just to stop using the search controller for the dashboard - my primary reason was performance & reducing use of a complex & poorly understood code construct but it also helps here

Note the search controller is something that we should aim to deprecate from everywhere except direct searches IMHO & then potentially either replace there or leap- by -extension away from it